universal-application-tool-0.0.1/app/repository/QuestionRepository.java

```diff
@@ -33,6 +33,57 @@ public QuestionRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exec
         executionContext);
   }
 
+  static class PathConflictDetector {
+    Boolean conflict = Boolean.FALSE;
+    String newPath;
+
+    public PathConflictDetector(String newPath) {
+      this.newPath = checkNotNull(newPath);
+    }
+
+    public Boolean hasConflict() {
+      return conflict;
+    }
+
+    public boolean checkConflict(Question question) {
+      boolean proceed = true;
+      if (pathConflicts(question.getPath(), newPath)) {
+        conflict = Boolean.TRUE;
+        proceed = false;
+      }
+      return proceed;
+    }
+
+    public static boolean pathConflicts(String path, String otherPath) {
+      return path.startsWith(otherPath) || otherPath.startsWith(path);
+    }
+  }
+
+  public CompletionStage<Boolean> lookupPathConflict(String newPath) {
+    return supplyAsync(
+        () -> {
+          PathConflictDetector detector = new PathConflictDetector(newPath);
+          ebeanServer.find(Question.class).findEachWhile(detector::checkConflict);
+          return detector.hasConflict();
+        },
+        executionContext);
+  }
+
+  public CompletionStage<Optional<Question>> lookupQuestionByPath(String path) {
+    return supplyAsync(
+        () ->
+            Optional.ofNullable(
+                ebeanServer
+                    .find(Question.class)
+                    .where()
+                    .eq("path", path)
+                    .orderBy()
+                    .desc("version")
```

Comment:
Suggestion (blocking): we have implemented versioning yet, I recommend removing this code for now since it may cause confusion

Reply:
Done.